lib/commands/selectByValue.js

```diff
@@ -34,38 +34,10 @@
  *
  */
 
-import { CommandError } from '../utils/ErrorHandler'
+import selectByAttribute from './selectByAttribute'
 
 let selectByValue = function (selectElem, value) {
-    /**
-     * convert value into string
-     */
-    if (typeof value === 'number') {
-        value = value.toString()
-    }
-
-    /*!
-     * parameter check
-     */
-    if (typeof selectElem !== 'string' || typeof value !== 'string') {
-        throw new CommandError(`number or type of arguments don't agree with selectByValue command`)
-    }
-
-    /**
-     * get options element by xpath
-     */
-    return this.element(selectElem).then((res) => {
-        /**
-         * find option elem using xpath
-         */
-        var normalized = '[normalize-space(@value) = "' + value.trim() + '"]'
-        return this.elementIdElement(res.value.ELEMENT, `./option${normalized}|./optgroup/option${normalized}`)
-    }).then((res) => {
-        /**
-         * select option
-         */
-        return this.elementIdClick(res.value.ELEMENT)
-    })
+  return selectByAttribute.call(this, selectElem, 'value', value)
```

Comment:
the original function uses `this.element` and complains if it doesn't rebind `this`.